android-o/AutofillFramework/AutofillFramework/MultiplePartitionsActivity.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Android.App;
+using Android.Content;
+using Android.OS;
+using Android.Runtime;
+using Android.Support.V7.App;
+using Android.Views;
+using Android.Views.Autofill;
+using Android.Widget;
+using Java.Lang;
+
+namespace AutofillFramework
+{
+    [Activity(Label = "MultiplePartitionsActivity")]
+    [Register("com.xamarin.AutofillFramework.MultiplePartitionsActivity")]
+    public class MultiplePartitionsActivity : AppCompatActivity
+    {
+        private ScrollableCustomVirtualView mCustomVirtualView;
+        private AutofillManager mAutofillManager;
+
+        private CustomVirtualView.Partition mCredentialsPartition;
+        private CustomVirtualView.Partition mCcPartition;
+
+        public static Intent GetStartActivityIntent(Context context)
+        {
+            var intent = new Intent(context, typeof(MultiplePartitionsActivity));
+            return intent;
+        }
+
+        protected override void OnCreate(Bundle savedInstanceState)
+        {
+            base.OnCreate(savedInstanceState);
+            SetContentView(Resource.Layout.multiple_partitions_activity);
+
+            mCustomVirtualView = FindViewById<ScrollableCustomVirtualView>(Resource.Id.custom_view);
+
+
+            mCredentialsPartition =
+                mCustomVirtualView.AddPartition(GetString(Resource.String.partition_credentials));
+            mCredentialsPartition.AddLine("username", (int) AutofillType.Text,
+                GetString(Resource.String.username_label),
+                "         ", false, View.AutofillHintUsername);
+            mCredentialsPartition.AddLine("password", (int) AutofillType.Text,
+                GetString(Resource.String.password_label),
+                "         ", true, View.AutofillHintPassword);
+
+            var ccExpirationType = (int) AutofillType.Date;
+            // TODO: add a checkbox to switch between text / date instead
+            Intent intent = Intent;
+            if (intent != null)
+            {
+                int newType = intent.GetIntExtra("dateType", -1);
+                if (newType != -1)
+                {
+                    ccExpirationType = newType;
+                    var typeMessage = GetString(Resource.String.message_credit_card_expiration_type,
+                        CommonUtil.GetTypeAsString(ccExpirationType));
+                    // TODO: display type in a header or proper status widget
+                    Toast.MakeText(ApplicationContext, typeMessage, ToastLength.Long).Show();
+                }
+            }
+
+            mCcPartition = mCustomVirtualView.AddPartition(GetString(Resource.String.partition_credit_card));
+            mCcPartition.AddLine("ccNumber", (int) AutofillType.Text,
```

Comment:
We can remove this cast once the `AddLine` signature has been changed to accept an `AutofillType`.